### PR TITLE
improve bytes feature extraction

### PR DESCRIPTION
**Pull request by mr-tz**
- reduces `FF FF FF FF ...` bytes sequences extracted by IDA
- small viv optimization

### CODE REVIEW

File: capa/features/extractors/ida/helpers.py [hunk 1]
```diff
@@ -167,6 +167,9 @@ def basic_block_size(bb):
 def read_bytes_at(ea, count):
     """ """
     segm_end = idc.get_segm_end(ea)
+    if segm_end == idaapi.BADADDR:
+        # address not in any segment
+        return
```
Comment (williballenthin) [inline capa/features/extractors/ida/helpers.py:172]: lets either raise and exception or return an empty bytes here. as is, it returns `None`, which differs from the other type returned by this function.

File: capa/features/extractors/ida/insn.py [hunk 2]
```diff
@@ -148,6 +148,9 @@ def extract_insn_bytes_features(f, bb, insn):
     example:
         push    offset iid_004118d4_IShellLinkA ; riid
     """
+    if idaapi.is_call_insn(insn):
+        return
```
Comment (williballenthin) [inline capa/features/extractors/ida/insn.py:152]: this one is ok cause its a generator, so just early return

File: capa/features/extractors/viv/insn.py [hunk 3]
```diff
@@ -258,10 +258,10 @@ def extract_insn_bytes_features(f, bb, insn):
     example:
         #     push    offset iid_004118d4_IShellLinkA ; riid
     """
-    for oper in insn.opers:
-        if insn.mnem == "call":
-            continue
+    if insn.mnem == "call":
+        return
```
Comment (williballenthin) [inline capa/features/extractors/viv/insn.py:262]: good fix

**Review by williballenthin — CHANGES_REQUESTED** [review]
one nit, otherwise looks great!

File: capa/features/extractors/ida/helpers.py [hunk 4]
```diff
@@ -167,6 +167,9 @@ def basic_block_size(bb):
 def read_bytes_at(ea, count):
     """ """
     segm_end = idc.get_segm_end(ea)
+    if segm_end == idaapi.BADADDR:
+        # address not in any segment
+        return 0
```
Comment (mr-tz) [inline capa/features/extractors/ida/helpers.py:172]: is this better?

**Comment by mr-tz** [comment]
adding a test as well, now

File: capa/features/extractors/ida/helpers.py (hunk 4 above)
Comment (mike-hunhoff) [reply capa/features/extractors/ida/helpers.py:172]: Let's return `b""` here to maintain a consistent return type.

**Comment by mr-tz** [comment]
```
...
OK   mimikatz-function=0x40105D-bytes(6E 6F 70 65)-False
OK   mimikatz-function=0x44570F-bytes(FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF)-False
OK   mimikatz-function=0x44EDEF-bytes(49 00 4E 00 50 00 55 00 54 00 45 00 56 00 45 00 4E 00 54 00)-True
OK   mimikatz-function=0x410DFC-characteristic(nzxor)-True
...
```

File: capa/features/extractors/ida/helpers.py (hunk 4 above)
Comment (mr-tz) [reply capa/features/extractors/ida/helpers.py:171]: ah, good point, let me address that... next week ;)

**Review by mike-hunhoff — COMMENTED** [review]
Love these improvements! Minor change for `read_bytes_at` otherwise this is great.

**Pull request merged**